File: sem5/labAlg/aula7-ex1.cpp
#include <iostream>
#include <vector>

#define ll long long int

using namespace std;

int main()
{
	ll n, x;
	int mod = 1e9+7;
	cin >> n >> x;
	vector<ll> coins(n);
	for(int i=0; i<n; i++)
	{
		cin >> coins[i];
	}
	vector<ll> dp(x+1, 0);
	
	dp[0] = 1;
	
	for(ll i=1; i<=x; i++)
	{
		for(ll j=0; j<n; j++)
		{
			if(i-coins[j] >= 0)
			{
				dp[i] += dp[i-coins[j]];
				dp[i] %= mod;
			}
		}
	}
	cout << dp[x] << endl;
	return 0;
}

File: sem5/labAlg/aula7-ex1.py




def money_sum(coin, n, x):
    sol = [0 for i in range(x+1)]
    mod = 1e9+7
    sol[0] = 1
    for i in range(1, x+1):
        for j in range(n):
            if i >= coin[j]:
                sol[i] += (sol[i-coin[j]])%mod

    return int(sol[x]%mod)

if __name__ == '__main__':
    # n = number of coins
    # x = desired sum
    n, x = map(int, input().split())
    coin = [int(x) for x in input().split()]
    print(money_sum(coin, n, x))
